abnf/odata-aggregation-testcases.yaml

```diff
@@ -333,7 +333,24 @@ TestCases:
     Rule: queryOptions
     Input: $apply=aggregate(Product/Name with Custom.concat as ProductNames)
     Expect:
-      - aggregatableFactor:Product/Name # is also an aggrPathPrimitive
+      - aggregatableFactor:Product/Name
+
+  - Name: aggregate - custom aggregation method with type cast
+    Rule: queryOptions
+    Input: $apply=aggregate(Products/Self.DigitalProduct with Custom.discount as Stuff)
+    Expect:
+      - aggrPathPrefix:Products/Self.DigitalProduct
+
+  - Name: aggregate - forbidden type cast after single-valued segment
+    Rule: queryOptions
+    Input: $apply=aggregate(Product/Self.DigitalProduct with Custom.discount as Stuff)
+    Expect:
+      - aggrPathPrefix:Product/Self.DigitalProduct
+
+  - Name: aggregate - custom aggregation method with only type cast forbidden
+    Rule: queryOptions
+    FailAt: 36
+    Input: $apply=aggregate(Self.DigitalProduct with Custom.discount as Stuff)
```

Comment:
TC 2022-03-01: Allow this (in analogy with "custom aggregate reached via type cast only").